### PR TITLE
Stale and others

### PR DESCRIPTION
**Pull request by cpanato**
Add the initial replacement for the stale bot and other small changes

tested in my local instance and worked fine over the weekend

note: we need some big refactor in the mattermod code, but for now this is not in the scope

### CODE REVIEW

File: server/pull_request.go [hunk 1]
```diff
@@ -266,6 +266,57 @@ func isSpinmintDoneComment(message string, upgrade bool) bool {
 	pattern := regexp.MustCompile(spinmintDoneMessage)
 	return pattern.MatchString(message)
 }
+
+func CheckPRActivity() {
+	mlog.Info("Checking if need to Stale a Pull request")
+	var prs []*model.PullRequest
+	if result := <-Srv.Store.PullRequest().ListOpen(); result.Err != nil {
```
Comment (hanzei) [inline server/pull_request.go]: Nit: You can drop the else and be compliant with Idiomatic Error Handling.

File: config/config-mattermod.json [hunk 2]
```diff
@@ -22,24 +22,27 @@
         }
     ],
 
+    "DaysUntilStale": 14,
+    "ExemptStaleLabels": [],
+    "StaleLabel": "",
+    "StaleComment": "",
```
Comment (hanzei) [inline config/config-mattermod.json]: :+1: on making them configurable.

File: server/pull_request.go [hunk 3]
```diff
@@ -266,6 +266,57 @@ func isSpinmintDoneComment(message string, upgrade bool) bool {
...
+			mlog.Error("Error getting Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number))
+		}
+
+		timeToStale := time.Now().AddDate(0, 0, -Config.DaysUntilStale)
+		if timeToStale.After(*pull.UpdatedAt) || timeToStale.Equal(*pull.UpdatedAt) {
+			var prLabels []string
+			canStale := true
+			if labels, _, err := client.Issues.ListLabelsByIssue(pr.RepoOwner, pr.RepoName, pr.Number, nil); err != nil {
+				mlog.Error("Error getting the labels in the Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number))
+			} else {
```
Comment (hanzei) [inline server/pull_request.go]: Same here. You can drop the else.

File: server/pull_request.go [hunk 4]
```diff
@@ -266,6 +266,57 @@ func isSpinmintDoneComment(message string, upgrade bool) bool {
...
+		mlog.Error(result.Err.Error())
+	} else {
+		prs = result.Data.([]*model.PullRequest)
+	}
+
+	client := NewGithubClient()
+	for _, pr := range prs {
+		pull, _, errPull := client.PullRequests.Get(pr.RepoOwner, pr.RepoName, pr.Number)
+		if errPull != nil {
+			mlog.Error("Error getting Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number))
```
Comment (hanzei) [inline server/pull_request.go]: I think you want to `break` here.

File: server/pull_request.go [hunk 5]
```diff
@@ -266,6 +266,57 @@ func isSpinmintDoneComment(message string, upgrade bool) bool {
...
+						break
+					}
+				}
+			}
+
+			if canStale {
+				label := []string{Config.StaleLabel}
+				_, _, errLabel := client.Issues.AddLabelsToIssue(pr.RepoOwner, pr.RepoName, pr.Number, label)
+				if errLabel != nil {
+					mlog.Error("Error adding the stale labe in the  Pull Request", mlog.String("RepoOwner", pr.RepoOwner), mlog.String("RepoName", pr.RepoName), mlog.Int("PRNumber", pr.Number))
```
Comment (hanzei) [inline server/pull_request.go]: I think you want to break here.

File: config/config-mattermod.json [hunk 6]
```diff
@@ -22,24 +22,27 @@
...
-
     "SetupSpinmintUpgradeTag": "",
     "SetupSpinmintUpgradeMessage": "",
     "SetupSpinmintUpgradeDoneMessage": "",
-
+    "BuildMobileAppTag": "",
+    "BuildMobileAppInitMessage": "",
+    "BuildMobileAppDoneMessage": "",
+    "BuildMobileAppFailedMessage": "",
+    "StartLoadtestTag": "",
```
Comment (hanzei) [inline config/config-mattermod.json]: Are these related to the PR topic?

File: server/pull_request.go (hunk 1 above)
Comment (cpanato) [reply server/pull_request.go]: i think is done.

File: config/config-mattermod.json (hunk 2 above)
Comment (cpanato) [reply config/config-mattermod.json]: at least for now

File: config/config-mattermod.json (hunk 6 above)
Comment (cpanato) [reply config/config-mattermod.json]: no, was missing

File: server/pull_request.go (hunk 5 above)
Comment (cpanato) [reply server/pull_request.go]: done

File: server/pull_request.go (hunk 4 above)
Comment (cpanato) [reply server/pull_request.go]: done

File: server/pull_request.go (hunk 3 above)
Comment (cpanato) [reply server/pull_request.go]: done
